### PR TITLE
[Flink AI Flow][Client] Add the ready check for the channel of AIFlowClient

### PR DESCRIPTION
**Pull request by lisy09**
## What is the purpose of the change

To fix the bug related to run_tests.sh & grpc UNAVAILABLE mentioned in #480.

## Brief change log

- add wait_for_ready_and_throw_error() in AIFlowClient
- in get_ai_flow_client(), call AIFlowClient.wait_for_ready_and_throw_error() for the case client is reused from _default_ai_flow_client
- remove sleep 2 in run_tests.sh

## Verifying this change 

This change is already covered by existing tests, such as *(please describe tests)*.

### CODE REVIEW

**Comment by lisy09** [comment]
As discussed with @SteNicholas, will move grpc.channel_ready_future logic to AIFlowClient.__init__(), and remove all modification to server side and test case.
Though it will brings behaviour change and impack quite large scale, it can benifit more as it fails faster in the client side when server not ready.

**Comment by lisy09** [comment]
@SteNicholas This pr is ready. Please help to check.

File: flink-ai-flow/ai_flow/client/ai_flow_client.py [hunk 1]
```diff
@@ -148,6 +154,21 @@ def __init__(self,
...
+            try:
+                channel = grpc.insecure_channel(uri)
+                fut = grpc.channel_ready_future(channel)
+                fut.result(self.CLIENT_INIT_WAIT_READY_TIMEOUT)
+                available = True
+                break
+            except:
+                pass
+        if not available:
+            raise Exception(f"Server {self.server_uri} not ready!")
```
Comment (SteNicholas) [inline flink-ai-flow/ai_flow/client/ai_flow_client.py:170]:
```suggestion
            raise Exception(f"Client connection to server({self.server_uri}) is not ready. Please confirm the status of the process for `AIFlowServer`.")
```

**Review by SteNicholas — CHANGES_REQUESTED** [review]

File: flink-ai-flow/ai_flow/client/ai_flow_client.py (hunk 1 above)
Comment (lisy09) [reply flink-ai-flow/ai_flow/client/ai_flow_client.py]: fixed with https://github.com/alibaba/flink-ai-extended/pull/484/commits/bd81f30ef6402bcd9e940ae0e63177d343de0224

**Review by SteNicholas — APPROVED** [review]

**Pull request merged**